### ISSUE TITLE
[BUG] Explicit examples for Open API query parameters do not use the defined "style"

### ISSUE THREAD

**Issue by m-develops** [schemathesis/schemathesis#1190]
**Describe the bug**
With the component below the query params are not rendered correctly.
The example below gives the uri: uri: 'http://localhost/api/product/v1/products?page=number&page=size'

**To Reproduce**
1. Make a GET path with the component below in the parameter
2. Run the schemathesis CLI command

If possible, please post a minimal version of your API schema that cause this behavior:

```yaml
components:
  parameters:
    page:
      in: query
      name: page
      description: Page number
      required: false
      style: deepObject
      explode: true
      schema:
        type: object
        additionalProperties: false
        properties:
          number:
            type: integer
            minimum: 1
            maximum: 100
            example: 1
          size:
            type: integer
            minimum: 1
            maximum: 100
            example: 15
```

**Expected behavior**
I would expect an uri like:  'http://localhost/api/product/v1/products?page[number]=1&page[size]=15'

**Additional context**
It looks like this worked in the past:
https://github.com/schemathesis/schemathesis/issues/599

**Comment by Stranger6667** [comment]
Hi @m-develops,

Thank you for reporting! Unfortunately, I can't reproduce it with the schema you provided.

```python
import requests
import schemathesis
from hypothesis import given, settings

SCHEMA = {
    "openapi": "3.0.2",
    "info": {
        "title": "Test",
        "description": "Test",
        "version": "0.1.0"
    },
    "paths": {
        "/data": {
            "get": {
                "parameters": [
                    {
                        "$ref": "#/components/parameters/page"
                    }
                ],
                "responses": {
                    "200": {
                        "description": "OK"
                    }
                }
            }
        }
    },
    "components": {
        "parameters": {
            "page": {
                "in": "query",
                "name": "page",
                "description": "Page number",
                "required": False,
                "style": "deepObject",
                "explode": True,
                "schema": {
                    "type": "object",
                    "additionalProperties": False,
                    "properties": {
                        "number": {
                            "type": "integer",
                            "minimum": 1,
                            "maximum": 100,
                            "example": 1,
                        },
                        "size": {
                            "type": "integer",
                            "minimum": 1,
                            "maximum": 100,
                            "example": 15,
                        }
                    }
                }
            }
        }
    }
}

schema = schemathesis.from_dict(SCHEMA)

@given(case=schema["/data"]["get"].as_strategy())
@settings(max_examples=25)
def test(case):
    request = requests.PreparedRequest()
    request.prepare(
        method="GET",
        url="http://localhost",
        params=case.query,
    )
    print(request.url)
```

Running the above with the current Schemathesis version via `pytest -s test_example.py` outputs the following:

```
http://localhost/
http://localhost/
http://localhost/
http://localhost/?page=
http://localhost/?page%5Bsize%5D=15&page%5Bnumber%5D=30
http://localhost/?page%5Bnumber%5D=1
http://localhost/?page%5Bnumber%5D=93&page%5Bsize%5D=2
http://localhost/?page%5Bnumber%5D=55
http://localhost/?page%5Bnumber%5D=55&page%5Bsize%5D=16
http://localhost/?page%5Bnumber%5D=98
http://localhost/?page%5Bnumber%5D=98&page%5Bsize%5D=32
http://localhost/?page%5Bnumber%5D=32&page%5Bsize%5D=32
http://localhost/?page%5Bnumber%5D=66
http://localhost/?page%5Bnumber%5D=66&page%5Bsize%5D=73
http://localhost/?page%5Bsize%5D=58&page%5Bnumber%5D=20
http://localhost/?page%5Bsize%5D=58&page%5Bnumber%5D=58
http://localhost/?page%5Bsize%5D=58&page%5Bnumber%5D=1
http://localhost/?page%5Bsize%5D=2
http://localhost/?page%5Bsize%5D=2&page%5Bnumber%5D=7
http://localhost/?page%5Bsize%5D=7&page%5Bnumber%5D=7
http://localhost/?page%5Bsize%5D=7&page%5Bnumber%5D=1
http://localhost/?page%5Bsize%5D=26&page%5Bnumber%5D=55
http://localhost/?page%5Bsize%5D=26&page%5Bnumber%5D=26
http://localhost/?page%5Bsize%5D=26
http://localhost/?page%5Bsize%5D=100
```

Which corresponds to urlencoded variants of URLs similar to what you provided in the "Expected behavior" section. I also observe the same behavior via Schemathesis CLI running against a JSON file of the schema above.

How did you find those invalid URLs? Was it CLI or Python tests? Did I miss something important in the schema? What Schemathesis / Python / Hypothesis versions are you using?

**Comment by m-develops** [comment]
Hi @Stranger6667,

Thanks for the quick response.
It looks related to using the `--checks all` with the cli.

When i am using the command below the result depends on using `--checks all` or not:
`schemathesis run --hypothesis-max-examples=10 --store-network-log=test-results/schemathesis-log.yaml --base-url=http://localhost/api specification/dummy-api.yaml`

Network log with option `--checks all`, here only the uri `http://localhost/api/data?page=number&page=size` is generated:
[schemathesis-log-with-checks-all.log](https://github.com/schemathesis/schemathesis/files/6574179/schemathesis-log-with-checks-all.log)

Network log without option `--checks all`, this still has the uri `http://localhost/api/data?page=number&page=size` but also others like you mentioned:
[schemathesis-log-without-checks-all.log](https://github.com/schemathesis/schemathesis/files/6574191/schemathesis-log-without-checks-all.log)

Yaml file i used:
```openapi: 3.0.2
info:
  title: Test
  description: Test
  version: 0.1.0
paths:
  /data:
    get:
      parameters:
        - $ref: '#/components/parameters/page'
      responses:
        '200':
          description: OK
components:
  parameters:
    page:
      in: query
      name: page
      description: Page number
      required: false
      style: deepObject
      explode: true
      schema:
        type: object
        additionalProperties: false
        properties:
          number:
            type: integer
            minimum: 1
            maximum: 100
            example: 1
          size:
            type: integer
            minimum: 1
            maximum: 100
            example: 15
```

**Comment by Stranger6667** [comment]
Thank you for providing more info! The problem is that explicit examples are not transformed to the proper representation. For the schema above, Schemathesis generates `Case(query={'page': {'number': 1, 'size': 15}})` as an explicit example (it goes first in the request log). But it should be `Case(query={'page[size]': 1, 'page[number]': 15})` instead

**Comment by Stranger6667** [comment]
The fix is released in `3.7.7`. Feel free to reopen the issue if it doesn't fix your case :)

**Comment by Stranger6667** [comment]
Thank you again for reporting the issue!
